src/app.js

```diff
@@ -1,2 +1,254 @@
+
+                if (sortDesc)
+                    products.reverse(); // <- This is why we're utilizing a vanillajs array.
+            }
+
+            return products;
+        },
+
+        // products displayed per page - This is a number. It considers the items-per-page UI
+        //                               selector and how many new products we can show on screen.
+        productsDisplayedPerPage: function () {
+
+            // Cache references to our watched properties.
+            const itemsPerPage = parseInt(this.get('itemsPerPage'));
+            const newProducts = this.get('newProducts');
+
+            // Determine how many existing products we can show per page.
+            let productsPerPage = itemsPerPage - newProducts.length;
+            if (productsPerPage < 1) {
+                while (newProducts.length >= itemsPerPage) {
+                    newProducts.pop();
+                }
+                productsPerPage = 1;
+            }
+            return productsPerPage;
+        },
+
+        // pages - This populates the page selector dropdown. It's just an array of integers. It 
+        //         regenerates based on how many sorted-and-filtered and new products we have. Items per page too.
+        pages: function () {
+
+            // Cache references to our watched properties.
+            const sortedAndFilteredProductsLength = this.get('sortedAndFilteredProducts.length');
+            const productsDisplayedPerPage = this.get('productsDisplayedPerPage');
+
+            // Determine how many pages we'll need and create an array of that length.
+            const numberOfPages = Math.ceil(sortedAndFilteredProductsLength / productsDisplayedPerPage);
+            let pages = [];
+            for (let i=0; i<numberOfPages; i++) {
+                pages[i] = i+1;
+            }
+            return pages;
+        },
+
+        // visible products - This is a vanilla array of Product Models. Only those that are present
+        //                    in the UI based on sorting, filtering, and paging.
+        visibleProducts: function () {
+
+            // Cache references to our watched properties.
+            const pageIdx = this.get('pageIdx');
+            const products = this.get('sortedAndFilteredProducts');
+            const productsDisplayedPerPage = this.get('productsDisplayedPerPage');
+
+            // Compute the start and end indexes.
+            const startIdx = pageIdx * productsDisplayedPerPage;
+            const endIdx = startIdx + productsDisplayedPerPage;
+
+            // Unfortunately, the Backbone Adaptor does not work with computed properties. :'(
+            // We still want to use the Collection.toJSON method though because our 'products' are Backbone Models.
+            return new ProductCollection( products.slice(startIdx, endIdx) ).toJSON();
+        },
+    },
+
+    oninit: function () {
+
+        // Request product information.
+        this.get('products').fetch();
```

Comment:
As soon as our application initializes, we make an HTTP request to get our product data. This is done via our Backbone Model extension, [Product Model](https://github.com/IAmBrandonMcGregor/mcgregor-ftproject/pull/1/files#diff-88cd8dfd6178461363230b4d6151d1c3).